examples/source_examples/mastertax/connector.py

```diff
@@ -0,0 +1,263 @@
+
+        return status, resource_id
+
+def get_process_status(url: str, headers: dict):
+    """
+    Gets the status and output_id for submitted process specified in URL
+    output_id is used to download files in the next step.
+
+    :param url: The URL to which the API request is made, containing resource_id .
+    :param headers: A dictionary of headers for authorization
+    :return: status and output_id
+    """
+    response = rq.get(url, headers=headers, cert=(cert_path, key_path))
+    if response.status_code != 200:
+        log.info(str(response))
+    response.raise_for_status()  # Ensure we raise an exception for HTTP errors.
+    response_page = response.json()
+    status = response_page.get("processingJob").get("processingJobStatusCode")
+    output_id = response_page.get("processingJob").get("processOutputID")
+
+    return status, output_id
+
+def download_file(url: str, headers: dict, name: str):
+    """
+    Downloads zip file with specified name
+    :param url: The URL to which the API request is made, containing ID of file to download
+    :param headers: A dictionary of headers for authorization
+    :param name: name of file to create
+    :return:
+    """
+    headers["Range"] = "bytes=0-"
+    response = rq.get(url, headers=headers, stream=True, cert=(cert_path, key_path))
+    if response.status_code != 200:
+        log.info(str(response))
+    response.raise_for_status()  # Ensure we raise an exception for HTTP errors.
+
+    with open(f"{name}.zip", "wb") as file:
+        for chunk in response.iter_content(chunk_size=1024):
+            file.write(chunk)
+
+    log.info("ZIP file downloaded successfully!")
+
+def make_headers(conf: dict):
+    """
+    Create authentication headers.
+
+    :param conf: Dictionary containing authentication details.
+    :param state: Dictionary storing token and expiration details.
+    :return: headers
+    """
+
+    url = "https://api.adp.com/auth/oauth/v2/token"
+    write_to_file(conf["crtFile"], cert_path)
+    write_to_file(conf["keyFile"], key_path)
+    payload = f"grant_type=client_credentials&client_id={conf['clientId']}&client_secret={conf['clientSecret']}"
+    headers = {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'Accept': 'application/json'
+    }
+
+    # Request a new token
+    try:
+        auth_response = rq.request("POST", url, headers=headers, data=payload,
+                                   cert=("SSL.crt", "SSL_auth.key"))
+        auth_response.raise_for_status()
+        auth_page = auth_response.json()
+
+        # Extract token safely
+        auth_token = auth_page.get("access_token")
+
+        if not auth_token:
+            raise ValueError("Authentication failed: accessToken missing in response")
+
+        return {"Authorization": f"Bearer {auth_token}", "Content-Type": "application/json"}
+
+    except rq.exceptions.RequestException as e:
+        raise RuntimeError(f"❌ Failed to authenticate: {e}")

```

Comment:
these characters can sometime not render properly on CLI, so let's stick with standard text.
```suggestion
        raise RuntimeError(f"Failed to authenticate: {e}")

```